pkg/resource/deploy/plan_executor.go

```diff
@@ -133,17 +133,19 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 	//     should bail.
 	//  3. The stepExecCancel cancel context gets canceled. This means some error occurred in the step executor
 	//     and we need to bail. This can also happen if the user hits Ctrl-C.
-	canceled, err := func() (bool, error) {
+	canceled, res := func() (bool, *result.Result) {
 		logging.V(4).Infof("planExecutor.Execute(...): waiting for incoming events")
 		for {
 			select {
 			case event := <-incomingEvents:
-				logging.V(4).Infof("planExecutor.Execute(...): incoming event (nil? %v, %v)", event.Event == nil, event.Error)
+				logging.V(4).Infof("planExecutor.Execute(...): incoming event (nil? %v, %v)", event.Event == nil, event.Result)
 
-				if event.Error != nil {
-					pe.reportError("", event.Error)
+				if event.Result != nil {
 					cancel()
-					return false, event.Error
+					if event.Result.Error() != nil {
+						pe.reportError("", event.Result.Error())
+					}
```

Comment:
another 'bail' check.

Reply:
once reported, we should convert results into Bails.